src/inverseRobot.js

```diff
@@ -25,6 +25,24 @@
  */
 function inverseRobot(robot) {
   // write code here
+  let reversed = {};
+  const arrayOfProp = [];
+
+  for (const prop in robot) {
+    arrayOfProp.push(robot[prop]);
+
+    if (robot.hasOwnProperty(prop)) {
+      reversed[robot[prop]] = prop;
+    }
+  }
+
+  for (const elem of arrayOfProp) {
+    if (arrayOfProp.indexOf(elem) !== arrayOfProp.lastIndexOf(elem)) {
+      reversed = null;
+    }
+  }
```

Comment:
read about hasOwnProperty

Reply:
Got it. I have used hasOwnProperty where it wasn't actually necessary.